src/axom/sina/docs/sphinx/core_concepts.rst

```diff
@@ -0,0 +1,25 @@
+.. ## Copyright (c) 2017-2024, Lawrence Livermore National Security, LLC and
+.. ## other Axom Project Developers. See the top-level LICENSE file for details.
+.. ##
+.. ## SPDX-License-Identifier: (BSD-3-Clause)
+
+=============
+Core Concepts
+=============
+
+Sina provides four main classes:
+
+    - ``Document`` - represents the top-level object of a JSON file conforming to the Sina schema.
+    - ``Record`` - represents the data to be stored.
+    - ``Relationship`` - represents a way to define the relationship between two ``Record`` objects.
+    - ``CurveSet`` - a class to store related independent and dependent ``Curve`` objects.
```

Comment:
An immediate question I had when reading this is whether/how `Curve`s and `Record`s are related

Reply:
Per @HaluskaR: "Curves are just one more type of data. So a record can have a few different types of data (scalars, strings, curve[ sets], library data...). If a record represents one run of a code, curves are some of the time and cycle plots generated while that code runs."

I'll put something to try to clarify this in the documentation :)